docs/docs/getting-started/android-setup.md

```diff
@@ -1 +1,236 @@
-# Android Setup
\ No newline at end of file
+# Android Setup
+
+After you've completed the [Rust setup](docs/getting-started/rust-setup.md) you should be able to run `cargo run --bin android`. This will create a new folder called `MoproAndroidBindings`. It should look like the structure
+
+```sh
+MoproAndroidBindings
+├── jniLibs
+│   ├── arm64-v8a
+│   │   └── libuniffi_mopro.so
+│   ├── armeabi-v7a
+│   │   └── libuniffi_mopro.so
+│   ├── x86
+│   │   └── libuniffi_mopro.so
+│   └── x86_64
+│       └── libuniffi_mopro.so
+└── uniffi
+    └── mopro
+        └── mopro.kt
+```
+
+First we will create an android app through Android Studio. If you already have an app project, you can skip this step. We'll do File -> New -> New Project and create an Empty Activity.
+
+![create an android app project](/img/android-example-1.png)
+
+Your android project should be opened now.
+
+Then add jna to `app/build.gradle.kts`
+
+```kts
+dependencies {
+   ...
+   implementation("net.java.dev.jna:jna:5.13.0@aar")
+   ...
+}
+```
+
+![add jna dependency](/img/android-example-2.png)
+
+Sync gradle with `shift`+`cmd`+`O`, or press
+
+![android sync gradle](/img/android-example-4.png)
+
+Open Finder and drag folders:
+
+1. Move the `MoproAndroidBindings/jniLibs/` folder into `app/src/main/jniLibs/`.
+2. Move the `MoproAndroidBindings/uniffi/mopro/mopro.kts` file into `app/src/main/java/uniffi/mopro/mopro.kt`
+
+![android bindings](/img/android-bindings.png)
+
+Create an asset folder: File -> New -> Folder -> Assets Folder. <br/>
+Paste the zkey in the assets folder.
+
+![android paste zkey](/img/android-example-3.png)
+
+## Proving from the app
+
+In your project, there should be a file named `MainActivity.kt`
+
+> It should be under `app/src/main/java/com/example/YOUR_APP/MainActivity.kt`
+
+Import the following functions:
+
+```kotlin
+import android.content.Context
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.ui.unit.dp
+import androidx.compose.material3.Button
+import kotlinx.coroutines.launch
+import uniffi.mopro.generateCircomProof
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+```
+
+This will make the proving functions `generateCircomProof` available in this module and also help to load zkey.
+
+In the `MainActivity.kt`, replace the `Greeting("Android")` with
+
+```kotlin
+MainScreen(this)
```

Comment:
A bit more context on where this goes would help. My project didn't have a `Greeting("Android")` line, it had more variables. Maybe like "make your `setContent` function look like this:
```kotlin
setContent {
          MoproAndroidAppTheme {
              // A surface container using the 'background' color from the theme
              Surface(
                  modifier = Modifier.fillMaxSize(),
                  color = MaterialTheme.colorScheme.background
              ) {
                  MainScreen(this)
              }
          }
      }
```